src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -175,11 +176,233 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
+        private BoundPattern BindSlicePattern(
+            SlicePatternSyntax node,
+            TypeSymbol inputType,
+            bool permitDesignations,
+            bool hasErrors,
+            bool misplaced,
+            BindingDiagnosticBag diagnostics)
+        {
+            if (misplaced && !hasErrors)
+            {
+                diagnostics.Add(ErrorCode.ERR_MisplacedSlicePattern, node.Location);
+                hasErrors = true;
+            }
+
+            MethodSymbol? sliceMethod = null;
+            BoundIndexerAccess? indexerAccess = null;
+
+            TypeSymbol sliceType;
+            if (inputType.IsSZArray())
+            {
+                sliceType = inputType;
+            }
+            else if (TryFindRangeIndexerPatternForListPattern(node, inputType, out sliceMethod, out indexerAccess))
+            {
+                sliceType = sliceMethod is not null ? sliceMethod.ReturnType :
+                            indexerAccess is not null ? indexerAccess.Type :
+                            throw ExceptionUtilities.Unreachable;
+            }
+            else
+            {
+                if (!inputType.IsErrorType())
+                    diagnostics.Add(ErrorCode.ERR_UnsupportedTypeForSlicePattern, node.Location, inputType.ToDisplayString());
+                hasErrors = true;
+                sliceType = CreateErrorType();
+            }
+
+            BoundPattern? pattern = node.Pattern is not null
+                ? BindPattern(node.Pattern, sliceType, ExternalScope, permitDesignations, hasErrors, diagnostics)
+                : null;
+
+            return new BoundSlicePattern(node, sliceMethod, indexerAccess, sliceType: sliceType, pattern, inputType: inputType, narrowedType: inputType, hasErrors);
+        }
+
+        private ImmutableArray<BoundPattern> BindListPatternSubpatterns(
+            PropertyPatternClauseSyntax node,
+            TypeSymbol inputType,
+            TypeSymbol elementType,
+            bool permitDesignations,
+            ref bool hasErrors,
+            out bool sawSlice,
+            BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(node.IsKind(SyntaxKind.ListPatternClause));
+
+            sawSlice = false;
+            var subpatterns = node.Subpatterns;
+            var builder = ArrayBuilder<BoundPattern>.GetInstance(subpatterns.Count);
+            foreach (SubpatternSyntax subpat in subpatterns)
+            {
+                // Any name would have make this a PropertyPatternClause
+                Debug.Assert(subpat.NameColon is null);
+
+                PatternSyntax pattern = subpat.Pattern;
+                BoundPattern boundPattern;
+                if (pattern is SlicePatternSyntax slice)
+                {
+                    if (sawSlice)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_MisplacedSlicePattern, slice.Location);
```

Comment:
Rather than separately report this here, we should just pass `sawSlice` as `misplaced` below.